partiql-cli/src/main/kotlin/org/partiql/cli/utils/ServiceLoaderUtil.kt

```diff
@@ -92,20 +92,35 @@ import org.partiql.value.stringValue
 import org.partiql.value.structValue
 import org.partiql.value.symbolValue
 import org.partiql.value.timeValue
+import java.io.File
 import java.math.BigDecimal
 import java.math.BigInteger
+import java.net.URLClassLoader
 import java.util.ServiceLoader
 import java.util.concurrent.locks.ReentrantLock
 import kotlin.concurrent.withLock
 
+/**
+ * A util class to load pluggable functions by invoking Java Service Loader.
+ */
 class ServiceLoaderUtil {
     companion object {
         private val lock = ReentrantLock()
 
         @OptIn(PartiQLFunctionExperimental::class)
         @JvmStatic
-        fun loadFunctions(): List<ExprFunction> = lock.withLock {
-            val plugins = ServiceLoader.load(Plugin::class.java)
+        fun loadFunctions(pluginPath: String): List<ExprFunction> = lock.withLock {
+            val pluginsDir = File(pluginPath)
+            val pluginFolders = pluginsDir.listFiles { file -> file.isDirectory }.orEmpty()
+            val files = pluginFolders.flatMap { folder ->
+                folder.listFiles { file -> file.isFile && file.extension == "jar" }.orEmpty().toList()
+            }
+            val plugins = if (files.isNotEmpty()) {
+                val classLoader = URLClassLoader.newInstance(files.map { it.toURI().toURL() }.toTypedArray())
+                ServiceLoader.load(Plugin::class.java, classLoader)
+            } else {
+                ServiceLoader.load(Plugin::class.java)
+            }
```

Comment:
For now, we can remove the `else`.

Reply:
I don't think we can remove the `else`. Because the default file path is "~/.partiql/plugins" and so the files might be empty.

Reply:
Since the CLI is an application, people don't use this utility directly.

Therefore, since we are the only users of `ServiceLoaderUtil` and since we don't provide any implementations of `Plugin`, we will never make use of the statement within the `else` block.

The only time we load `Plugins` is when external customers provide their implementations of `Plugin` -- which is through the filesystem.